services/src/main/java/org/keycloak/protocol/oidc/grants/AuthorizationCodeGrantType.java

```diff
@@ -127,24 +130,32 @@ public Response process(Context context) {
         }
 
         if (redirectUri != null && !redirectUri.equals(redirectUriParam)) {
-            event.error(Errors.INVALID_CODE);
-            logger.tracef("Parameter 'redirect_uri' did not match originally saved redirect URI used in initial OIDC request. Saved redirectUri: %s, redirectUri parameter: %s", redirectUri, redirectUriParam);
+            event.error(Errors.INVALID_REDIRECT_URI);
```

Comment:
Here I used the more appropriate `Errors.INVALID_REDIRECT_URI` instead of `Errors.INVALID_CODE`